modules/core/04-channel/types/packet.go

```diff
@@ -35,6 +55,51 @@ func CommitPacket(cdc codec.BinaryCodec, packet Packet) []byte {
 	return hash[:]
 }
 
+// CommitEurekaPacket returns the Eureka packet commitment bytes. The commitment consists of:
+// sha256_hash(timeout_timestamp + timeout_height.RevisionNumber + timeout_height.RevisionHeight + sha256_hash(data))
+// from a given packet. This results in a fixed length preimage.
+// NOTE: A fixed length preimage is ESSENTIAL to prevent relayers from being able
+// to malleate the packet fields and create a commitment hash that matches the original packet.
+// NOTE: sdk.Uint64ToBigEndian sets the uint64 to a slice of length 8.
+func CommitEurekaPacket(packet Packet) []byte {
+	timeoutHeight := packet.GetTimeoutHeight()
+
+	timeoutBuf := sdk.Uint64ToBigEndian(packet.GetTimeoutTimestamp())
+
+	// only hash the timeout height if it is non-zero. This will allow us to remove it entirely in the future
```

Comment:
If we want to use the fixed-size preimage, which I think is a good thing, can we use here a sort of "padding" bytes to reach the fixed-size when the height is zero?